sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinationsHelpers.java

```diff
@@ -299,19 +304,42 @@ static class ConstantTimePartitioningClusteringDestinations<T>
       this.jsonClustering = jsonClustering;
     }
 
+    boolean isJsonConfigPresent(String json) {
```

Comment:
this helper method can be static, and taking a ValueProvider<String> as parameter for further simplification?